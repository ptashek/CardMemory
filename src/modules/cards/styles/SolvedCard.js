export default (theme) => ({
  avatar: {
    position: 'absolute',
    marginTop: '20px',
    marginLeft: '20px',
    border: '3px solid black',
    fontWeight: 700,
    color: theme.palette.white,
    backgroundColor: theme.palette.colors.coconut600,
    pointerEvents: 'none',
  },
});
