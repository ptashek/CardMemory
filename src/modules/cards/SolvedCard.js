// @flow
import type { ComponentType } from 'react';
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Avatar from '@material-ui/core/Avatar';
import colors from '@workday/canvas-colors-web';
import styles from './styles/SolvedCard';

const useStyles = makeStyles(styles);

type SolvedCardProps = {
  faceColor: $Keys<typeof colors>,
  faceText: string,
};

const SolvedCard: ComponentType<SolvedCardProps> = (props: SolvedCardProps) => {
  const classes = useStyles();

  return (
    <Box
      data-testid="solved-card"
      borderRadius={16}
      boxShadow={3}
      width={80}
      height={80}
      m={2}
      bgcolor={`colors.${props.faceColor}`}
    >
      <Avatar className={classes.avatar}>{props.faceText}</Avatar>
    </Box>
  );
};

export default React.memo<SolvedCardProps>(SolvedCard);
