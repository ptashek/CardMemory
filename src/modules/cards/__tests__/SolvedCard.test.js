import React from 'react';
import { render } from '@testing-library/react';
import SolvedCard from '../SolvedCard';

describe('<SolvedCard />', () => {
  test('matches snapshot', () => {
    const { container } = renderWithTheme(<SolvedCard faceColor="cinnamon300" faceText="42" />);
    expect(container).toMatchSnapshot();
  });

  test('renders the passed in face text', () => {
    const { getByText } = renderWithTheme(<SolvedCard faceColor="cinnamon300" faceText="42" />);
    const avatarElement = getByText('42');
    expect(avatarElement).toBeInTheDocument();
  });
});
