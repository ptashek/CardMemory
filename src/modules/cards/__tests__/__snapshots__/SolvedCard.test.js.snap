// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SolvedCard /> matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
    data-testid="solved-card"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-1 MuiAvatar-colorDefault"
    >
      42
    </div>
  </div>
</div>
`;
